crates/oxc_linter/src/rules/unicorn/prefer_optional_catch_binding.rs

```diff
@@ -51,7 +51,29 @@ impl Rule for PreferOptionalCatchBinding {
         if references_count != 0 {
             return;
         }
-        ctx.diagnostic(prefer_optional_catch_binding_diagnostic(catch_param.pattern.span()));
+        let Some(parent_node) = ctx.nodes().parent_node(node.id()) else {
+            return;
+        };
+        let AstKind::CatchClause(catch_clause) = parent_node.kind() else {
+            return;
```

Comment:
i wonder wether there should be some debug assertions here
(basically assert that they're unreachable) as a `CatchParameter` should always:
 - have a parent node
 - have a `CatchClause` ast kind as the parent

Reply:
cc @DonIsaac

Reply:
> i wonder wether there should be some debug assertions here (basically assert that they're unreachable) as a `CatchParameter` should always:
> 
> * have a parent node
> * have a `CatchClause` ast kind as the parent

Oh, this is just to get the value of catch_cause. Is there a better way to get it?

Reply:
i think this way is fine, but the `else` clauses here are unreachable. lets see what Don says 🙂

Reply:
we could also rewrite this rule to be something like:
```rs
let AstKind::CatchClause(catch_clause) = node.kind() else { return; }
let Some(catch_param) = catch_clause.param else {return}

let references_count = get_param_references_count(&catch_param.pattern, ctx);
if references_count != 0 {
  return;
}

// report diagnostic
```

Reply:
> i wonder wether there should be some debug assertions here (basically assert that they're unreachable) as a `CatchParameter` should always:
> 
> * have a parent node
> * have a `CatchClause` ast kind as the parent

These kinds of debug assertions act as tests on Semantic. I'm generally in favor of them.